src/UiDatepicker.vue

```diff
@@ -328,7 +338,7 @@ export default {
 
         onPickerOpen() {
             if (this.usesModal) {
-                this.valueAtModalOpen = this.value ? dateUtils.clone(this.value) : null;
+                this.valueAtModalOpen = this.dateObject ? this.dateObject : null;
```

Comment:
Here we cloned the value because we don't want the copy we're referencing to change when `this.value` or `this.dateObject` is mutated.

Any reason why you removed the clone?